test/mod/test_util_mnt.py

```diff
@@ -20,3 +21,28 @@ def test_mount_guard_failure_msg(tmp_path):
             mg.mount("/dev/invalid-src", tmp_path)
     assert "special device /dev/invalid-src does not exist" in str(e.value)
 
+
+# This needs a proper refactor so that FileSystemMountService just uses
+# a common mount helper.
+class TestFileSystemMountService(FileSystemMountService):
+    def __init__(self, args=None):
+        # override __init__ to make it testable
+        pass
+    def translate_options(self, options):
+        return options
+
+
+@pytest.mark.skipif(os.getuid() != 0, reason="root only")
+def test_osbuild_mount_failure_msg(tmp_path):
+    mnt_service = TestFileSystemMountService()
+    # yes, we have a third way of mounting things
+    with pytest.raises(RuntimeError) as e:
+        args = {
+            "source": "/dev/invalid-src",
+            "target": os.fspath(tmp_path),
+            "root": "/",
+            "options": [],
+        }
+        mnt_service.mount(args)
+    assert "special device /dev/invalid-src does not exist" in str(e.value)
```

Comment:
This should be inside the with block to guarantee `e`.

Reply:
Nice catch, the other two existing tests also do this (slightly) wrong.  I fixed this now

Reply:
Actually no it shouldn't be. This is the correct way to handle this.

Reply:
ok, I will just drop the commit then.